src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -112,28 +114,26 @@ public void launch(SlaveComputer computer, TaskListener listener) {
             Pod pod = template.build(slave);
             slave.assignPod(pod);
 
-            String podId = pod.getMetadata().getName();
+            String podName = pod.getMetadata().getName();
 
             String namespace = Arrays.asList( //
                     pod.getMetadata().getNamespace(),
                     template.getNamespace(), client.getNamespace()) //
                     .stream().filter(s -> StringUtils.isNotBlank(s)).findFirst().orElse(null);
             slave.setNamespace(namespace);
 
-            LOGGER.log(Level.FINE, "Creating Pod: {0}/{1}", new Object[] { namespace, podId });
+            LOGGER.log(Level.FINE, "Creating Pod: {0}/{1}", new Object[] { namespace, podName });
             pod = client.pods().inNamespace(namespace).create(pod);
-            LOGGER.log(INFO, "Created Pod: {0}/{1}", new Object[] { namespace, podId });
-            listener.getLogger().printf("Created Pod: %s/%s%n", namespace, podId);
+            LOGGER.log(INFO, "Created Pod: {0}/{1}", new Object[] { namespace, podName });
+            listener.getLogger().printf("Created Pod: %s/%s%n", namespace, podName);
 
             TaskListener runListener = template.getListener();
-            runListener.getLogger().printf("Created Pod: %s in namespace %s%n", podId, namespace);
-
+            runListener.getLogger().printf("Created Pod: %s/%s%n", namespace, podName);
 
-            String podName = pod.getMetadata().getName();
-            String namespace1 = pod.getMetadata().getNamespace();
             template.getWorkspaceVolume().createVolume(client, pod.getMetadata());
             watcher = new AllContainersRunningPodWatcher(client, pod, runListener);
-            try (Watch _w = client.pods().inNamespace(namespace1).withName(podName).watch(watcher)) {
+            try (Watch w1 = client.pods().inNamespace(namespace).withName(podName).watch(watcher);
+                 Watch w2 = client.events().inNamespace(namespace).withField("involvedObject.name", podName).watch(new TaskListenerEventWatcher(podName, runListener))) {
                 watcher.await(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
             } catch (InvalidDockerImageException e) {
```

Comment:
Is this special case still needed? I am not really sure what it was doing to begin with; looks suspicious.

Reply:
It fails the build if one of the docker images is invalid (https://github.com/jenkinsci/kubernetes-plugin/pull/497). Ideally I think it would deserve a rework but it works.

Reply:
But the build would fail anyway, without this clause; the exception would be thrown up out of `launch`. Right? And with this PR, you would see the appropriate diagnostics right in the log. But the current code here does more, by canceling some queue items (which seem to be _any_ `node` blocks associated with this build, which is wrong by the way—should only be canceling the `node` block associated with this pod). If that is needed in order to prevent the build from hanging, then it is needed in many other error cases too. So this does not add up for me.

Reply:
No, the launcher would just fail but this wouldn't affect the build so the build would wait forever.

> should only be canceling the node block associated with this pod
probably.

Reply:
I do not think this is anything blocking the current PR but it looks like something to revisit.

Reply:
@jglick I'm revisiting this, not sure how to find out item should be cancelled exactly.
The only correlation I can make seems to be based on label, since at this point the queue item is not yet assigned to the node. There could be several items waiting for a node if they are inside a parallel block or using a static pod template. So, cancel the first one only?

Reply:
Not sure.